src/layer-model.js

```diff
@@ -1,12 +1,28 @@
 import isEqual from 'lodash/isEqual';
 
+// decodes
+import canvasDecodes from './canvas-decodes';
+
 class LayerModel {
   opacity = 1;
 
   visibility = true;
 
   constructor(layerSpec = {}) {
-    Object.assign(this, layerSpec, { changedAttributes: {} });
+    // parse params before setting model's properties
+    Object.assign(this, {
```

Comment:
Why not ```{ ...this, ...layer ... } ``` etc etc

Reply:
Something like `this = {...this, ...layer };` is not allowed, so I won't touch this by now.